tutorials/groups.qmd

```diff
@@ -0,0 +1,242 @@
+
+# OR, the preferred generalized get_coordinate function
+xθi_ = get_coordinates(G, p0, xXi_, B)[1]
+_xθi_ = get_coordinates(M, p0, xXi_, B)[1]
+
+# confirm all versions are correct
+@assert isapprox(xθi, xθi_); @assert isapprox(xθi, _xθi_)
+@assert isapprox(xθi, xθi__); @assert isapprox(xθi, _xθi__)
+```
+
+::: {.callout-note}
+The disadvantage might be that the representation of `X` is not nice, i.e. it uses too much space or doing vector-calculations is not so easy.  E.g. fixed rank matrices are overloaded for all vector operations, but maybe that is “not enough” for a general user application that really wants vectors. But: Given a basis `B` one can look at the coefficients of the tangent vector `X` with respect to basis `B`.  From the Sphere example note above the basis vectors would be `Y1=[0.0, 1.0, 0.0]` and `Y2=[0.0, 0.0, 1.0]`, so to get the coordinates would be `c = get_coordinates(Sphere(2), p, X, B)`.  Vice versa, if you have a coordinate vector with respect to a basis `B` of the tangent space at `p` and want the vector back, then you do `X2 = get_vector(M, p, c, B)` (and you should have `X2==X`).  The coordinate vector `c` might also have the advantage of saving memory. For example SPD matrix tangent vectors take $n^2$ entries to save, i.e. storing the full matrix, but the coordinate vectors only take $\frac{n(n+1)}{2}$.
+:::
+
+
+### Actions and Operations
+
+With the basics in hand on how to move between the coordinate, algebra, and group representations, let's briefly look at composition and application of points on the manifold.  For example, a `Rotations(n)` manifold is the mathematical representation, but the points have an application purpose in retaining information regarding a specific rotation.
+
+Points from a Lie group may have an associated action (i.e. a rotation) which we [`apply`](@ref).  Consider rotating through `θ = π/6` three vectors `V` from their native domain `Euclidean(2)`, from the reference point `a` to a new point `b`. Engineering disciplines sometimes refer to the action of a manifold point `a` or `b` as reference frames. More generally, by taking the tangent space at point `p`, we are defining a local coordinate frame with basis `B`, and should not be confused with "reference frame" `a` or `b`.
+
+Keeping with our two-dimensional example above:
+```{julia}
+aV1 = [1; 0]
+aV2 = [0; 1]
+aV3 = [10; 10]
+
+A_left = RotationAction(Euclidean(2), G)
+
+bθa = π/6
+bXa = get_vector(base_manifold(G), p0, bθa, B)
+
+bRa = exp(G, p0, bXa)
+
+for aV in [aV1; aV2; aV3]
+    bV = apply(A_left, bRa, aV)
+    # test we are getting the rotated vectors in Euclidean(2) as expected
+    @assert isapprox(bV[1], norm(aV) * cos(bθa))
+    @assert isapprox(bV[2], norm(aV) * sin(bθa))
+end
+```
+
+::: {.callout-note}
+In general, actions are usually non-commutative and the user must therefore be weary of [`LeftAction`](@ref) or [`RightAction`](@ref) needs. As in this case, the default `LeftAction()` is used.
```

Comment:
what does “weary” mean here? Also from thereon the grammar of the sentence seems to be broken. I do not understand this sentence.

Reply:
I've rephrased it.